migrations/versions/e2be4ab896d3_migrate_old_labels.py

```diff
@@ -0,0 +1,149 @@
+    column('project_id', sa.Integer()),
+    column('technical_level', sa.Unicode(40)),
+    column('session_type', sa.Unicode(40)),
+    column('section_id', sa.Integer()),
+    )
+
+project = table('project',
+    column('id', sa.Integer()),
+    column('name', sa.Unicode(250))
+    )
+
+section = table('section',
+    column('id', sa.Integer()),
+    column('project_id', sa.Integer()),
+    column('name', sa.Unicode(250)),
+    column('title', sa.Unicode(250)),
+    column('description', sa.UnicodeText()),
+    column('created_at', sa.DateTime()),
+    column('updated_at', sa.DateTime())
+    )
+
+label = table('label',
+    column('id', sa.Integer()),
+    column('name', sa.Unicode(250)),
+    column('title', sa.Unicode(250)),
+    column('description', sa.UnicodeText()),
+    column('icon_emoji', sa.UnicodeText()),
+    column('project_id', sa.Integer()),
+    column('main_label_id', sa.Integer()),
+    column('seq', sa.Integer()),
+    column('restricted', sa.Boolean()),
+    column('required', sa.Boolean()),
+    column('archived', sa.Boolean()),
+    column('created_at', sa.DateTime()),
+    column('updated_at', sa.DateTime())
+    )
+
+proposal_label = table('proposal_label',
+    column('proposal_id', sa.Integer()),
+    column('label_id', sa.Integer()),
+    column('created_at', sa.DateTime())
+    )
+
+
+def upgrade():
+    conn = op.get_bind()
+    # Migrate sections -
+    projects = conn.execute(project.select())
+    for proj in projects:
+        sec_count = conn.scalar(
+                sa.select([sa.func.count('*')]).select_from(section).where(section.c.project_id == proj['id'])
+            )
+        if sec_count > 0:
+            # the project has some sections
+            # create section labelset for the project
+            labset = conn.execute(label.insert().values({
+                    'project_id': proj['id'], 'name': u"section", 'title': u"Section",
+                    'seq': 1, 'description': u"", 'restricted': False, 'archived': False,
+                    'required': True, 'created_at': datetime.now(), 'updated_at': datetime.now()
+                }).returning(label.c.id)).first()
+
+            sections = conn.execute(section.select().where(section.c.project_id == proj['id']))
+            for index, sec in enumerate(sections, start=1):
+                lab = conn.execute(label.insert().values({
+                    'main_label_id': labset[0],
+                    'project_id': proj['id'], 'name': sec['name'], 'title': sec['title'],
+                    'seq': index, 'description': u"", 'restricted': False, 'archived': False,
+                    'required': True, 'created_at': datetime.now(), 'updated_at': datetime.now()
+                }).returning(label.c.id, label.c.name)).first()
+
+                proposals = conn.execute(
+                        proposal.select().where(proposal.c.section_id == sec['id']).where(proposal.c.project_id == sec['project_id'])
+                    )
```

Comment:
This will miss proposals in a project linked to sections in another project -- an error condition resulting from proposals that have been moved around. Leave a comment indicating this happens. We'll come around to finding a solution before deploying.